src/lib/token_list_gen.ts

```diff
@@ -101,9 +111,28 @@ export const generateTokenList = async (
       .map((token, i: number) => {
     const l2GatewayAddress = token.token.joinTableEntry[0].gateway.gatewayAddr;
     let { name:_name, decimals, symbol:_symbol } = token.tokenDatum;
-    if(!_name) throw new Error(`Unexpected undefined token name: ${JSON.stringify(token)}`);
-    if(!decimals) throw new Error(`Unexpected undefined token decimals: ${JSON.stringify(token)}`);
-    if(!_symbol) throw new Error(`Unexpected undefined token symbol: ${JSON.stringify(token)}`);
+    
+    if(decimals === undefined) throw new Error(`Unexpected undefined token decimals: ${JSON.stringify(token)}`);
```

Comment:
Could _name, _decimals or _symbol be null? I dont know if they can, just pointing out that before we would throw an error for that whereas now we wouldnt

Reply:
it can't be null according to typescript inference

Reply:
the previous issue is that !decimals was handling the value zero as falsey in a token